regex-syntax/src/parser.rs

```diff
@@ -2424,6 +2489,14 @@ mod tests {
         ]));
     }
 
+    #[test]
+    fn ignore_space_escape_space() {
+        assert_eq!(p(r"(?x)a\  # hi there"), Expr::Concat(vec![
```

Comment:
Allowing to escape whitespace is in line with what PCRE2 does, see here: http://www.pcre.org/current/doc/html/pcre2pattern.html#SEC5

> An escaping backslash can be used to include a white space or # character as part of the pattern.

And it seems less surprising than allowing `\ d` or the following to mean digits:

```rust
r"(?x)\ # comment
    d"
```